### PR TITLE
ci: Parallelize `ci-master` (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Latest e2e run: https://github.com/n8n-io/n8n/actions/runs/5740542213

### CODE REVIEW

File: .github/workflows/ci-master.yml [hunk 1]
```diff
@@ -32,18 +32,15 @@ jobs:
...
-          CI_LINT_MASTER: true
-        run: pnpm lint
+          path: |
+            /github/home/.cache
+            /github/home/.pnpm-store
+            ./node_modules
+            ./packages
+          key: ${{ github.sha }}-base:18-test-lint
 
       - name: Notify Slack on failure
```
Comment (ivov) [inline .github/workflows/ci-master.yml:45]: Do we need to duplicate the Slack notification step or will it capture issues from parallel jobs?

**Review by netroy — COMMENTED** [review]
Can you please add `parallelize-ci-master` to `on.push.branches` to test if this works as expected ?

**Review by netroy — CHANGES_REQUESTED** [review]
This is now using the node-version matrix only for the build process, and not for the entire flow. which means that we aren't testing the builds on node 20.
If this is too complicated to fix, we can disable node 20 support for now.

**Comment by ivov** [comment]
> This is now using the node-version matrix only for the build process, and not for the entire flow. which means that we aren't testing the builds on node 20. If this is too complicated to fix, we can disable node 20 support for now.

I'll revisit as soon as I can! Need to focus on main task for now.

**Review by netroy — APPROVED** [review]

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.2.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.2.1)

**Pull request merged**